### PR TITLE
feat: petregister tests

### LINKED ISSUE #225: Teste para a pagina PetRegisterPage
**Describe the feature**
Criar testes para a pagina de cadastro de raça de Pet

**Additional Information**
 
  
[ Issue 93 da task ](https://github.com/devhatt/pet-dex-frontend/issues/93)
</details>

### PR DESCRIPTION
**Pull request by juliaam**
Closes #225 

<details open> 
  <summary>
    <b>Feature</b>
  </summary>

Criação de testes para a página de PetRegister

</details>

<details open> 
  <summary>
    <b>Visual evidences:

image

</b>
  </summary>

</details>

<details open> 
  <summary>
    <b>Checklist</b>
  </summary>

- [x] Issue linked
- [x] Build working correctly
- [x] Tests created
</details>

### CODE REVIEW

File: src/layouts/app/pages/PetRegister/index.spec.js [hunk 1]
```diff
@@ -1,9 +1,108 @@
...
+      const mixedBreedCard = screen.getByLabelText(/mixed breed/i);
+      await userEvent.click(mixedBreedCard);
+
+      const button = screen.getByRole('button');
+      await userEvent.click(button);
+
+      expect(callBackEmit).toHaveBeenCalledWith('Mixed Breed');
+    });
+
+    it('does not emit the form data when does not have card select', async () => {
```
[the remark this answers is not in the document]
Comment (Alecell) [reply src/layouts/app/pages/PetRegister/index.spec.js:97]: `does not happen when theres no card selected`

File: src/layouts/app/pages/PetRegister/index.spec.js [hunk 2]
```diff
@@ -1,9 +1,108 @@
...
+    await userEvent.click(mixedBreedCard);
+
+    await userEvent.click(afghanHoundCard);
+
+    expect(mixedBreedCard).not.toHaveClass('pet-container--active');
+    expect(afghanHoundCard).toHaveClass('pet-container--active');
+  });
+
+  describe('emits', () => {
+    it('emits the form data when every field is validated', async () => {
```
[the remark this answers is not in the document]
Comment (Alecell) [reply src/layouts/app/pages/PetRegister/index.spec.js:83]: `form data when every field is validated`

**Review by Alecell — CHANGES_REQUESTED** [review]

**Review by DominMFD — APPROVED** [review]
Ta amassando nos testes

File: src/layouts/app/pages/PetRegister/index.spec.js [hunk 3]
```diff
@@ -1,9 +1,123 @@
...
+      const mixedBreedCard = screen.getByLabelText(/mixed breed/i);
+      await userEvent.click(mixedBreedCard);
+
+      const button = screen.getByRole('button');
+      await userEvent.click(button);
+
+      expect(callBackEmit).toHaveBeenCalledWith('Mixed Breed');
+    });
+
+    it('does not happen when theres no card selected when theres no card selected', async () => {
```
Comment (Alecell) [inline src/layouts/app/pages/PetRegister/index.spec.js:112]: Ta escrito `when theres no card selected` duas vezes

**Review by Alecell — CHANGES_REQUESTED** [review]

File: src/layouts/app/pages/PetRegister/index.spec.js (hunk 3 above)
Comment (juliaam) [reply src/layouts/app/pages/PetRegister/index.spec.js:112]: DOUBLE CLICK

**Review by Alecell — APPROVED** [review]

**Pull request merged**